### ISSUE TITLE
Inconsistency in function TwoPhaseDensity()

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#501]
**Reported by msielemann on 17 Feb 2011 12:29 UTC**
In [browser:Modelica/trunk/Modelica/Fluid/Dissipation.mo#L10245 function TwoPhaseDensity()], the mean two phase density assuming **homogeneous** approach is returned if `massFlowRateCorrection == true`.

This contradicts the comments to `massFlowRateCorrection` (Consider **heterogeneous** mass flow rate correction). This is given at several places. 

I assume that the code should look like this:

```
function TwoPhaseDensity
  // (...)
algorithm 
  rho_2ph := if not massFlowRateCorrection then rho_hom else if 
    voidFractionApproach == Modelica.Fluid.Dissipation.Utilities.Types.VoidFractionApproach.Homogeneous then 
          rho_hom else if voidFractionApproach == Modelica.Fluid.Dissipation.Utilities.Types.VoidFractionApproach.Momentum then 
          rho_mom else if voidFractionApproach == Modelica.Fluid.Dissipation.Utilities.Types.VoidFractionApproach.Energy then 
          rho_kin else MIN;
end TwoPhaseDensity;
```

BTW, reading other comments such as 
```
//SOURCE_3: p.52, eq. 4.6: heterogenous effects on momentum pressure loss considered through corrected mass flow rate
```
I think that the comment could be improved to say something like "Consider heterogeneous mass flow rate correction **on momentum pressure loss**". This be correct, wouldn't it?

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/501

**Comment by modelica-trac-importer** [comment]
**Comment by wischhusen on 30 Mar 2011 18:48 UTC**
Yes, you're right.

**Comment by modelica-trac-importer** [comment]
**Comment by wischhusen on 11 Apr 2012 13:19 UTC**
Fixed in 2ebb84b14dd7dbbd32e5922b40be5dd3cb1e531b

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 11 Apr 2012 20:44 UTC**
